src/core/send.c

```diff
@@ -1150,6 +1150,30 @@ QuicSendFlush(
     }
     _Analysis_assume_(Builder.Metadata != NULL);
 
+    if (Builder.Path->EcnValidationState == ECN_VALIDATION_CAPABLE) {
+        Builder.EcnEctSet = TRUE;
+    } else if (Builder.Path->EcnValidationState == ECN_VALIDATION_TESTING) {
+        if (Builder.Path->EcnTestingEndingTimeSet) {
+            if (!CxPlatTimeAtOrBefore64(
+                    CxPlatTimeUs64(), Builder.Path->EcnTestingEndingTime)) {
+                Builder.Path->EcnValidationState = ECN_VALIDATION_UNKNOWN;
+                QuicTraceLogConnInfo(
+                    EcnValidationUnknown,
+                    Connection,
+                    "ECN unknown.");
+            }
+        } else {
+            uint32_t ThreePtosInUs =
+                QuicLossDetectionComputeProbeTimeout(
```

Comment:
This will always use the default initial RTT for an estimate. Is that ok?

Reply:
`ThreePtosInUs` for a new connection is 3s. I can cap it if you want.